R/task-settings.R

```diff
@@ -0,0 +1,153 @@
+#' @title TaskSettings
+#' @description  R6 class defining properties of plot task settings
+#' @export
+TaskSettings <- R6::R6Class(
+  "PlotTaskSettings",
+  public = list(
+    #' @description
+    #' Create a `TaskSettings` object
+    #' @param taskName name of the task using the settings
+    #' @return A new `TaskSettings` object
+    initialize = function(taskName) {
+      validateIsIncluded(taskName, AllAvailableTasks)
+
+      private$.bins <- reEnv$defaultBins
+      private$.stairstep <- reEnv$defaultStairstep
+      private$.digits <- reEnv$formatNumericsDigits
+      private$.nsmall <- reEnv$formatNumericsSmall
+      private$.scientific <- reEnv$formatNumericsScientific
+      private$.scales <- list(Linear = TRUE, Logarithmic = TRUE)
+      private$.plotConfigurations <- list()
+      private$.applicationRanges <- NULL
+
+      # Plot configurations fields depend on the task
+      if (isIncluded(taskName, AllAvailableTasks$plotAbsorption)) {
+        private$.plotConfigurations <- list(absorptionPlot = NULL)
+      }
+      if (isIncluded(taskName, AllAvailableTasks$plotMassBalance)) {
+        private$.plotConfigurations <- list(
+          timeProfile = NULL,
+          cumulativeTimeProfile = NULL,
+          normalizedTimeProfile = NULL,
+          normalizedCumulativeTimeProfile = NULL,
+          pieChart = NULL
+        )

```

Comment:
why is here a `PlotTimeProfilesAndResiduals`-configuration set for the MassBalanceTask?